modules/porous_flow/include/userobjects/PorousFlowDictator.h

```diff
@@ -0,0 +1,175 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
+#ifndef POROUSFLOWDICTATOR_H
+#define POROUSFLOWDICTATOR_H
+
+#include "GeneralUserObject.h"
+#include "Coupleable.h"
+#include "ZeroInterface.h"
+
+class PorousFlowDictator;
+
+template<>
+InputParameters validParams<PorousFlowDictator>();
+
+/**
+ * This holds maps between the nonlinear variables
+ * used in a PorousFlow simulation and the
+ * variable number used internally by MOOSE, as
+ * well as the number of fluid phases and
+ * the number of fluid components.
+ *
+ * All PorousFlow Materials and Kernels calculate
+ * and use derivatives with respect to all the variables
+ * mentioned in this Object, at least in principal
+ * (in practice they may be lazy and not compute
+ * all derivatives).
+ */
+class PorousFlowDictator :
+  public GeneralUserObject,
+  public Coupleable,
+  public ZeroInterface
+{
+ public:
+  PorousFlowDictator(const InputParameters & parameters);
+
+  void initialize();
+  void execute();
+  void finalize();
+
+  /**
+   * The number of PorousFlow variables.  Materials
+   * and Kernels will calculate and use derivatives
+   * with respect to these variables in the Jacobian
+   */
+  unsigned int num_variables() const;
```

Comment:
We usually use camelCase for member functions and underscore_separated for local variable names.